ocaml/idl/autogen/api-ref-autogen.md

```diff
@@ -0,0 +1,12 @@
+---
+ layout: doc
+---
+
+# API Reference
+
+Version **@xapi-version@**
```

Comment:
Is there an easy way to have this replaced during build time? If not, I can have the pipeline replace it later).

Reply:
Currently, the version is embedded into binaries using the `Xapi_version` module and using the `dune-build-info` library. One way to do it would be to it would be to create an ocaml binary that processes this file.

One way you can try is to use `dune subst`, and use `%%VERSION%%`. I believe it's intended to be used project-wide, but you're welcome ot try

Reply:
Thanks for the tip; I won't consider this as a blocker for this PR though (it's probably not a big change, but I only have time to work on this on and off, so I'd rather get the PR in the soonest possible and improve the versioning later).

Reply:
Had a look, and the binary that generated the files already has access to this library, it should as easy as generating the markdown file from a mustache template to fill in the version... but for some reason I keep getting 0.0.0.dev :(

https://github.com/xapi-project/xen-api/commit/55a970baf71e44571b1eff72018bbb100ab667e9

Reply:
As discussed, building in the CI yields `0.0.dev`. In the interest of avoiding taking up too much of people's time, I can always leave it as is and replace it later in the pipeline alongside `@root@`.